### PR TITLE
Autorun will choose correct device when launching application

### PR DESCRIPTION
**Pull request by todi1856**
### Purpose of PR
**Type of change:**
- [X] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe) 

* **Description of the feature**

Logcat was connecting to the wrong device, when there were two or more devices connected to PC. The issue was because Unity didn't provide any info on which device it runs the application.

OnPostrocessLaunch callback was introduced in this PR - https://github.cds.internal.unity3d.com/unity/unity/pull/56719

### Checklist for PR maker
- [ ] Have you added a backport label? (if needed)
- [ ] Have you updated the Changelog? _Each package has a `CHANGELOG.md` file_
- [ ] **Have you added or updated the Documentation to your PR?** _When you add a new feature, change a property name, or change the behaviour of a feature, it's best practice to include related documentation changes in the same PR_
---
### Testing status
* **Existing or new automation tests - what automation was added, changed**
N/A
* **Detailed description of what testing was done, what projects were run**
  * Connect two devices to PC
  * Build & Run on device A, see that logcat connects to device A
  * Build & Run on device B, see that logcat connects to device B

* **What environment did you test on?**
Windows, Android 10 and Android 14 connected at the same time.

### Note to guardian
_Changeset to use, if extra changesets on top. Any other notes to pass along_

### Comments to reviewers
_Notes for the reviewers you have assigned_

### CODE REVIEW

**Review by IGuscin — COMMENTED** [review]
Works as expected.
Tested with 2 devices connected, build and run to device A, device B, All compatible devices, Default device, Patch and Run cases. Logcat properly picks which device to connect to (the one app was deployed to, or in case of all compatible devices - first device app was installed to), log in the console provides details on app installation to the device and launch data.

No issues found, but postponing approval until callback PR lands to trunk

File: com.unity.mobile.android-logcat/Editor/AndroidLogcatAutoSelect.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        public void Reset()
+        {
+            Finish();
+        }
+
+        public override string ToString()
+        {
+            return $"State: {m_State}, DeviceId: {DeviceId}, Pacakge: {PackageName}";
```
Comment (BorisShir) [inline com.unity.mobile.android-logcat/Editor/AndroidLogcatAutoSelect.cs:75]: Package

**Review by IGuscin — APPROVED** [review]
Main PR landed, small concern regarding callback added when target platform is not Android addressed, good to go.

File: com.unity.mobile.android-logcat/Editor/AndroidLogcatAutoSelect.cs (hunk 1 above)
Comment (todi1856) [reply com.unity.mobile.android-logcat/Editor/AndroidLogcatAutoSelect.cs]: https://github.com/Unity-Technologies/com.unity.mobile.logcat/pull/193/commits/056466112a97d8fcd2d2aa44372a06c00622d506

**Review by BorisShir — APPROVED** [review]

**Pull request merged**